documentation/src/main/asciidoc/user_guide/integrations.adoc

```diff
@@ -11,6 +11,214 @@ Also check the link:$$https://github.com/infinispan/infinispan-spark/tree/master
+
+.Index write strategies
+* One node writes, the other delegate to it sending messages
+* Each node writes on turns
+* You application makes sure it will only ever apply index writes on one node
+
+The _Infinispan Lucene Directory_ protects its content by implementing a distributed locking strategy, though this is designed as a last line of defense and is not to be considered an efficient mechanism to coordinate multiple writes: if you don't apply one of the above suggestions and get high write contention from multiple nodes you will likely get timeout exception.
+
+==== Suggestions for optimal performance
+
+===== JGroups and networking stack
+JGroups manages all network IO and as such it is a critical component to tune for your specific environment. Make sure to read the link:$$http://jgroups.org/manual-3.x/html/index.html$$[JGroups reference documentation] , and play with the performance tests included in JGroups to make sure your network stack is setup appropriately. Don't forget to check also operating system level parameters, for example buffer sizes dedicated for networking. JGroups will log warning when it detects something wrong, but there is much more you can look into.
+
+===== Using a CacheStore
+Currently all CacheStore implementations provided by Infinispan have a significant slowdown; we hope to resolve that soon but for the time being if you need high performance on writes with the Lucene Directory the best option is to disable any CacheStore; the second best option is to configure the CacheStore as _async_ . If you only need to load a Lucene index from read-only storage, see the above description for _org.infinispan.lucene.cachestore.LuceneCacheLoader_ .
+
+===== Apply standard Lucene tuning
+All known options of Lucene apply to the Infinispan Lucene Directory as well; of course the effect might be less significant in some cases, but you should definitely read the link:$$http://lucene.apache.org/core/index.html$$[Apache Lucene documentation] .
+
+===== Disable batching and transactions
+Early versions required Infinispan to have batching or transactions enabled. This is no longer a requirement, and in fact disabling them should provide little improvement in performance.
+
+===== Set the right chunk size
+The chunk size can be specified using the link:$$https://docs.jboss.org/infinispan/{infinispanversion}/apidocs/org/infinispan/lucene/directory/DirectotyBuilder.html$$[DirectoryBuilder] fluent API. To correctly set this variable you need to estimate what the expected size of your segments is; generally this is trivial by looking at the file size of the index segments generated by your application when it's using the standard FSDirectory. You then have to consider:
+
+* The chunk size affects the size of internally created buffers, so you don't want an outrageously large array as you're going to waste precious JVM memory. Also consider that during index writing such arrays are frequently allocated.
```

Comment:
Maybe it was when this doc was written a few years ago :smile:
I will rephrase this paragraph